packages/svelte/src/compiler/phases/1-parse/state/tag.js

```diff
@@ -263,38 +265,58 @@ function open(parser) {
 		return;
 	}
 
-	if (parser.eat('snippet')) {
-		parser.require_whitespace();
+	if (parser.match('snippet')) {
+		const snippet_declaration_end = find_matching_bracket(parser.template, parser.index, '{');
 
-		const name_start = parser.index;
-		const name = parser.read_identifier();
-		const name_end = parser.index;
+		if (!snippet_declaration_end) {
+			error(parser.template.length, 'unexpected-eof');
+		}
 
-		parser.eat('(', true);
+		// we'll eat this later, so save it
+		const raw_snippet_declaration = parser.template.slice(start, snippet_declaration_end + 1);
+		const start_subtractions = '{#snippet ';
+		const end_subtractions = '}';
+
+		// now the snippet declaration is just a function declaration (`function snipName() {}`)
+		const snippet_declaration = `function ${raw_snippet_declaration.slice(
```

Comment:
is there a reason not to use the existing `read_context` function? (granted, it should probably be called `read_pattern`)

Reply:
Other than "it doesn't work", no 😂 I tried everything I could think of and had no luck, but by all means feel free to mess around with it.